src/manifolds/Circle.jl

```diff
@@ -105,6 +105,19 @@ distance(::Circle{ℝ}, p::Real, q::Real) = abs(sym_rem(p - q))
 distance(::Circle{ℝ}, p, q) = abs(sum(sym_rem.(p - q)))
 distance(::Circle{ℂ}, p, q) = acos(clamp(complex_dot(p, q), -1, 1))
 
+@doc raw"""
+    embed(M::Circle, p)
+
+Embed a point `p` on [`Circle`](@ref) `M` in the ambient space. It returns `p`.
+"""
+embed(::Circle, p) = p
+@doc raw"""
+    embed(M::Circle, p, X)
+
+Embed a tangent vector `X` at `p` on [`Circle`](@ref) `M` in the ambient space. It returns `X`.
+"""
+embed(::Circle, p, X) = X
```

Comment:
I thought this was the default, but if it is not, then sure this line is correct.

Reply:
No, the default is to call the mutating `embed!` and it fails for `Circle` when it uses immutable numbers.

Reply:
ah, right!